pages/news/models.py

```diff
@@ -0,0 +1,82 @@
+from django.db import models
+from django.utils.text import slugify
+from django.utils.safestring import mark_safe
+import markdown
+
+
+class News(models.Model):
+    """News model for adding portal updates.
+
+    Represents news items that can be updates, announcements, etc.
+    Each news can conntain the following information
+
+    Attributes:
+        title (str): Display name of the News (max 200 chars, unique).
+        slug (str): URL-friendly version of title (auto-generated).
+        summary (str): Brief summary to be displayed in news index page.
+        content (str): Rich markdown content for detail pages.
+        image (ImageField): Image to be used in index and detail page.
+        is_active (bool): Whether news is visible (default: True).
+        created_at (datetime): When news was created.
+        updated_at (datetime): When news was last updated.
+
+    Example:
+        Create a new News:
+
+        .. code-block:: python
+
+            news = News.objects.create(
+                title="New highlight added",
+                summary="A new data highlight was added...",
+                content="The data highlight is about...",
+                image="some_image.png"
+            )
+            # slug automatically generated as "new-highlight-added"
+    """
+
+    title = models.CharField(
+        max_length=200, unique=True, help_text="Title of news item"
+    )
+    slug = models.SlugField(
+        max_length=200,
+        unique=True,
+        help_text="URL-friendly version of the title (auto-generated from Title)",
+    )
+    summary = models.TextField(help_text="Short summary for news index page")
+    content = models.TextField(
+        help_text="Rich text content in markdown format (displayed on news detail page)"
+    )
+    image = models.ImageField(
+        upload_to="news/images/", help_text="Image to be used in index and detail page"
+    )
+    is_active = models.BooleanField(
+        default=True, help_text="Whether this news is active and visible"
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ["-created_at"]
+        verbose_name = "News"
+        verbose_name_plural = "News"
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
+        """Save the news, auto-generating slug if not provided."""
+        if not self.slug:
+            self.slug = slugify(self.name)
```

Comment:
Shouldn't it be:
```suggestion
            self.slug = slugify(self.title)
```

Anyway, I don't see when this block intervenes because if I create a news with no slug the submission is rejected

Reply:
Ahh ok, now I understand. Yes, thats the behaviour of django admin. If a field doesn't have `blank=True`, then it is considered as "required" field in form validation ([docs](https://docs.djangoproject.com/en/5.2/ref/models/fields/#blank)).

Like I mentioned before, the `save()` method is what used to save the object (by `Django admin` too, after form validation). So having this extra check will help in cases where the object is saved outside django admin (by a script for example), in such cases if the `slug` was not provided, it will be generated from title.